app/emoji.go

```diff
@@ -70,7 +73,13 @@ func (a *App) CreateEmoji(sessionUserId string, emoji *model.Emoji, multiPartIma
 
 	emoji, err := a.Srv().Store.Emoji().Save(emoji)
 	if err != nil {
-		return nil, err
+		var appErr *model.AppError
+		switch {
+		case errors.As(err, &appErr):
+			return nil, appErr
+		default:
+			return nil, model.NewAppError("createEmoji", "api.emoji.create.internal_error", nil, err.Error(), http.StatusBadRequest)
```

Comment:
`CreateEmoji`. Also return code should be internal error. And error key should be "app.emoji..."